tests/test_selection/test_drop_high_psi_features.py

```diff
@@ -0,0 +1,543 @@
+    data = df.copy()
+    data["var_3"].iloc[15] = np.nan
+
+    var_col = [col for col in data if "var" in col]
+
+    transformer = DropHighPSIFeatures(missing_values="ignore")
+    transformed = transformer.fit_transform(data)
+
+    pd.testing.assert_frame_equal(transformed, data[var_col])
+
+
+def test_raise_missing_value_inf(df):
+    """Test an error is raised for inf when missing values is set to raise."""
+    data = df.copy()
+    data["var_3"].iloc[15] = np.inf
+
+    with pytest.raises(ValueError):
+        transformer = DropHighPSIFeatures(missing_values="raise")
+        transformer.fit(data)
+
+
+# ========= tests for _split_dataframe() fit ====
+
+
+quantile_test = [(0.5, 50), (0.33, 33), (0.17, 17), (0.81, 81)]
+
+
+@pytest.mark.parametrize("split_frac, expected", quantile_test)
+def test_calculation_quantile(split_frac, expected):
+    """Test the calculation of the quantiles using distinct values."""
+    df = pd.DataFrame(
+        {"A": [it for it in range(0, 101)], "B": [it for it in range(0, 101)]}
+    )
+
+    test = DropHighPSIFeatures(
+        split_col="A", split_frac=split_frac, split_distinct=False
+    )
+    test.fit_transform(df)
+    assert test.cut_off_ == expected
+
+
+def test_calculation_distinct_value():
+    """Test the calculation of the quantiles using distinct values."""
+    df = pd.DataFrame(
+        {"A": [it for it in range(0, 30)], "C": ["A", "B", "C", "D", "D", "D"] * 5}
+    )
+
+    test = DropHighPSIFeatures(split_col="C", split_frac=0.5, split_distinct=False)
+
+    test.fit_transform(df)
+    assert test.cut_off_ == "C"
+
+    test = DropHighPSIFeatures(split_col="C", split_frac=0.5, split_distinct=True)
+    test.fit_transform(df)
+    assert test.cut_off_ == "B"
+
+
+def test_calculation_df_split_with_different_types(df_mixed_types):
+    """Test the split of the dataframe using different type of variables."""
+    results = {}
+    cut_offs = {}
+    for split_col in df_mixed_types.columns:
+        test = DropHighPSIFeatures(split_frac=0.5, split_col=split_col)
+        test.fit_transform(df_mixed_types)
+        results[split_col] = test.psi_values_
+        cut_offs[split_col] = test.cut_off_
+
+    assert results["A"] == pytest.approx({"B": 0.0}, 12)
+    assert results["B"] == pytest.approx({"A": 3.0375978817052403}, 12)
+    assert results["C"] == pytest.approx({"A": 2.27819841127893, "B": 0.0}, 12)
+    assert results["time"] == pytest.approx({"A": 8.283089355027482, "B": 0.0}, 12)
+
+    expected_cut_offs = {"A": 9.5, "B": 1.5, "C": "B", "time": date(2019, 1, 10)}
+
+    assert cut_offs == expected_cut_offs
+
+    # Test when no columns is defined
+    test = DropHighPSIFeatures(split_frac=0.5)
+    test.fit_transform(df_mixed_types)
+    assert test.psi_values_ == pytest.approx({"A": 8.283089355027482, "B": 0.0}, 12)
+
+
+def test_calculation_no_split_columns():
+    """Test the split of the dataframe using different type of variables."""
```

Comment:
I am still not clear what the aim of this test is. Is it to test `variables =None`? `split_col=None`? also, why do you set split_distinct to True for the test?

Could we add  the object(s) of the test in assert statements? This way I find it clearer to understand what the objective of the test is

Reply:
I see this test is redundant as it is included in the test just above it when doing 
![image](https://user-images.githubusercontent.com/32060943/140821960-8f89d885-b339-4067-826d-19459b7849a0.png)
The idea was to test string and date would not lead to an error when no split_column is defined.

I will remove the test.